libs/chonky/src/embedding/hugging_face_api.rs

```diff
@@ -0,0 +1,177 @@
+use std::path::Path;
+
+use error_stack::{Report, ResultExt as _};
+use reqwest::{
+    Client,
+    header::{AUTHORIZATION, CONTENT_TYPE, HeaderMap, HeaderValue},
+};
+use serde::Deserialize;
+use tokio::fs;
+
+use crate::ChonkyError;
+
+#[derive(Deserialize, Debug)]
+pub struct BoundingBox {
+    pub xmin: f32,
+    pub ymin: f32,
+    pub xmax: f32,
+    pub ymax: f32,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct TablePrediction {
+    pub score: f32,
+    #[serde(rename = "box")]
+    pub bounding_box: BoundingBox,
+}
+
+//for now just have it as environment variable
+fn get_hugging_face_token() -> Result<String, Report<ChonkyError>> {
+    std::env::var("HUGGING_FACE_TOKEN").change_context(ChonkyError::HuggingFaceAPI)
+}
+
+async fn get_binary_image_data(
+    image_path: impl AsRef<Path> + core::marker::Send + core::marker::Sync,
+) -> Result<Vec<u8>, Report<ChonkyError>> {
+    fs::read(image_path)
+        .await
+        .change_context(ChonkyError::ImageError)
+}
```

Comment:
This really doesn't more than reading a file. We should probably inline it. Also, the error `ImageError` seems to be inaccurate as this is really a file operation. (Btw, `tokio::fs::read` only wraps the reading operation in an another task, nothing else).

If we keep it, we should use `Send` only and directly:
```suggestion
async fn get_binary_image_data(
    image_path: impl AsRef<Path> + Send,
) -> Result<Vec<u8>, Report<ChonkyError>> {
    fs::read(image_path)
        .await
        .change_context(ChonkyError::ImageError)
}
```

Reply:
In trying to move the function inline, found that the explicit references to `Send + Sync` bounds avoid compiler errors when using references to `image_path` further downstream when retrying the request if the hugging face model was initially cold.